velox/substrait/SubstraitToVeloxPlanValidator.cpp

```diff
@@ -0,0 +1,336 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/substrait/SubstraitToVeloxPlanValidator.h"
+#include "TypeUtils.h"
+
+#include "velox/functions/prestosql/registration/RegistrationFunctions.h"
+
+namespace facebook::velox::substrait {
+
+bool SubstraitToVeloxPlanValidator::validate(
```

Comment:
Does this method need to match SubstraitParser::parseType? Is there a way to avoid divergence? Any particular reason other types are not supported?